apis/v1alpha2/shared_types.go

```diff
@@ -27,10 +27,6 @@ import (
 //
 // The API object must be valid in the cluster; the Group and Kind must
 // be registered in the cluster for this reference to be valid.
-//
-// References to objects with invalid Group and Kind are not valid, and must
```

Comment:
Feels like this is trying to say something like "if the reference is to an object you don't control, don't accept it", but that seems pretty redundant, agreed.